### ISSUE TITLE
Log error when no datastore compatible with specified storage policy was found

### ISSUE THREAD

**Issue by omniproc** [kubernetes-sigs/cluster-api-provider-vsphere#1317]
/kind feature

**Describe the solution you'd like**
When not specifying a datastore and using a `VSPHERE_STORAGE_POLICY` CAPV will currently not report any errors when it doesn't find a datastore compatible with the specified policy. Instead the depolyment will hang in a reconcile loop forever without any indicator that explains why the depolyment stopped.

Log an error when no compatible datastore was found.

**Environment:**

- Cluster-api-provider-vsphere version: 1.0.1

**Comment by gab-satchi** [comment]
Hey @omniproc, we do try and rely on conditions more for displaying error statuses. Logging can be useful for debugging but users shouldn't be required to look through logs, ideally. 

The error should be bubbled up into the `VMProvisioned` condition on the `VSphereVM` resource. It may also be mirrored onto the `VSphereMachine`.

**Comment by omniproc** [comment]
> The error should be bubbled up into the `VMProvisioned` condition on the `VSphereVM` resource. It may also be mirrored onto the `VSphereMachine`.

I didn't see an event there but I'll try to reproduce the issue and double check. Will let you know my findings when I know more.

**Comment by gab-satchi** [comment]
The conditions will be in the `Status` field of those resources. Won't be emitting events

**Comment by omniproc** [comment]
indeed, I can see the message in the condition of the resource. However CAPV has a couple of CRDs. Wouldn't it make sense to trigger events for certain error conditions of the CRDs? Most K8s tools will highlight / prompt the user when a error event occured which makes it more user friendly in my opinion.

**Comment by srm09** [comment]
Currently in CAPV, we have been using `Conditions` to bubble specific error scenarios which block operations on the CRDs. In this case, the corresponding condition gets eventually bubbled up to the CAPI Machine object too. For most of the other failures too, we are raising the conditions correctly which eventually get bubbled up to the corresponding/owner CAPI objects,
Plus emitting events for this particular scenario when the object gets reconciled would result in multiple events of similar kind being emitted for the same issue, which would clobber the events list.

**Comment by srm09** [comment]
A good way to about this would be to list out the error scenarios per controller and then take a stock of what errors does the community think would be useful to emit in terms of events. wdyt?

**Comment by omniproc** [comment]
I get the idea of conditions bubbleing up and events being triggered by the object the user actualy has controll over. However `the corresponding condition gets eventually bubbled up to the CAPI Machine object too` doesn't seem to be the case here. I never say any event informing me about the issue anywhere. So I had to look through _all_ CRDs conditions to find out what's causing the depolyment to stall. Is that how it's supposed to be?

It looks like the responsibility of what resource only "bubbled events up" and what resource actually fires events seems not be be defined strictly enough. I understand that you want to limit duplicated events but there seem to be no guarantees that a event will be triggered at all currently which doesn't look like the desired outcome. But maybe it's just me.
